torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -186,4 +186,121 @@ torch::CppFunction autogradNotImplementedFallback() {
+  const auto& arguments = schema.arguments();
+  const auto& returns = schema.returns();
+  const auto num_arguments = arguments.size();
+  const auto num_returns = returns.size();
+  const auto stack_start = stack->size() - num_arguments;
+  bool any_is_inplace = false;
+
+  at::Tensor aliased_input;
+
+  int64_t aliased_output_idx = -1;
+  for (const auto i : c10::irange(num_returns)) {
+    const auto& alias_info = returns[i].alias_info();
+    if (alias_info.has_value() && !alias_info->isWrite()) {
+      AT_ASSERT(
+        aliased_output_idx == -1,
+        "Expected only a single output in the operator schema to have a non-write alias annotation (i.e., 'Tensor(a)'). "
+        "Non-composite functions where multiple outputs are aliased with inputs aren't supported."
+        "Please rewrite your function as a composite function.");
+      aliased_output_idx = i;
+    }
+  }
+
+  int64_t aliased_input_idx = -1;
+  for (const auto i : c10::irange(num_arguments)) {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value()) {
+      if (!alias_info->isWrite()) {
+        AT_ASSERT(
+          aliased_input_idx == -1,
+          "Expected only a single input in the operator schema to have a non-write alias annotation (i.e., 'Tensor(a)'). "
+          "Non-composite functions where multiple inputs are aliased with outputs aren't supported. "
+          "Please rewrite your function as a composite function.");
+        aliased_input_idx = i;
+        const c10::IValue& aliased_input_iv = (*stack)[stack_start + i]; // get a reference to an ivalue on the stack
+        TORCH_CHECK(aliased_input_iv.isTensor());
+        // copy assignment
+        aliased_input = aliased_input_iv.toTensor();
+      } else {
+        any_is_inplace = true;
+      }
+    }
+  }
+  // NOTE: [ Limitations of ADInplaceOrView boxed kernel ]
+  // We can assume the view is always the first return, and that there will only be a single return
+  // TODO: Document the limitations of this kernel for custom op writers
+  TORCH_INTERNAL_ASSERT((aliased_input_idx == -1 && aliased_output_idx == -1) ||
+    (aliased_input_idx == 0 && aliased_output_idx == 0))
+  const bool is_view = aliased_input_idx != -1;
+  const bool need_view_func = is_view && !aliased_input.unsafeGetTensorImpl()->support_as_strided();
+
+  std::function<at::Tensor(const at::Tensor&)> view_func = nullptr;
+  std::vector<c10::IValue> stack_args_copy;
+  if (need_view_func) {
+    stack_args_copy = std::vector<c10::IValue>(stack->begin() + stack_start, stack->end());
+  }
+
+  if (need_view_func) {
+    view_func = [=](const at::Tensor& t) {
+      // - Maybe add a test for this...
+      // - See NOTE: [ Limitations of ADInplaceOrView boxed kernel ]
+      // - We have to make another copy because the same lambda can be used twice! e.g., double backward
+      std::vector<c10::IValue> stack_args_copy_copy(stack_args_copy);
+      stack_args_copy_copy.at(0) = t;
```

Comment:
If you're always going to override the 0th index, maybe don't record it in `stack_args_copy_copy` in the first place